File: frontend/src/lib/api/perkCard.ts
import apiClient from '@/lib/apiClient'

export interface PerkCardItem {
  id: string
  from_user_id: string
  to_user_id: string
  title: string
  description: string
  icon_type: string
  status: 'active' | 'used' | 'expired' | string
  used_at: string | null
  expires_at: string | null
  created_at: string
}

export interface CreatePerkCardRequest {
  title: string
  description?: string
  icon_type?: string
  expires_at?: string
}

export function createPerkCard(data: CreatePerkCardRequest): Promise<PerkCardItem> {
  return apiClient.post('/api/v1/perk-cards', data).then((r) => r.data)
}

export function getPerkCards(): Promise<PerkCardItem[]> {
  return apiClient.get('/api/v1/perk-cards').then((r) => r.data)
}

export function usePerkCard(id: string): Promise<PerkCardItem> {
  return apiClient.post(`/api/v1/perk-cards/${id}/use`).then((r) => r.data)
}

File: backend/internal/dto/perk_card.go
package dto

import "time"

type CreatePerkCardRequest struct {
	Title       string     `json:"title" binding:"required,min=1,max=100"`
	Description string     `json:"description" binding:"omitempty,max=500"`
	IconType    string     `json:"icon_type" binding:"omitempty,max=50"`
	ExpiresAt   *time.Time `json:"expires_at"`
}

type PerkCardItem struct {
	ID          string     `json:"id"`
	FromUserID  string     `json:"from_user_id"`
	ToUserID    string     `json:"to_user_id"`
	Title       string     `json:"title"`
	Description string     `json:"description"`
	IconType    string     `json:"icon_type"`
	Status      string     `json:"status"`
	UsedAt      *time.Time `json:"used_at"`
	ExpiresAt   *time.Time `json:"expires_at"`
	CreatedAt   time.Time  `json:"created_at"`
}
